recipes/effolkronium-random/all/conanfile.py

```diff
@@ -1,31 +1,49 @@
 import os
 
-from conans import ConanFile, tools
+from conan import ConanFile
+from conan.tools.files import copy, get
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=1.52.0"
+
 
 class RandomConan(ConanFile):
     name = "effolkronium-random"
     description = "Random for modern C++ with convenient API."
     license = "MIT"
-    topics = ("conan", "random", "header-only")
-    homepage = "https://github.com/effolkronium/random"
     url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/effolkronium/random"
+    topics = ("random", "header-only")
+
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("random-" + self.version, self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def package(self):
-        self.copy("LICENSE.MIT", dst="licenses", src=self._source_subfolder)
-        self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "include"))
-
-    def package_id(self):
-        self.info.header_only()
+        copy(self, "LICENSE.MIT", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(
+            self,
+            "*.hpp",
+            dst=os.path.join(self.package_folder, "include"),
+            src=os.path.join(self.source_folder, "include"),
+        )
 
     def package_info(self):
+        self.cpp_info.bindirs = []
+        self.cpp_info.libdirs = []
+
+        self.cpp_info.set_property("cmake_file_name", "effolkronium_random")
+        self.cpp_info.set_property("cmake_target_name", "effolkronium_random::effolkronium_random")
+
```

Comment:
Checking upstream: https://github.com/effolkronium/random/blob/master/CMakeLists.txt I think the correct target name is `effolkronium_random` not `effolkronium_random::effolkronium_random`

Reply:
Indeed: https://github.com/effolkronium/random/tree/master#cmake
Thanks! Fixed it.